src/gwt/src/org/rstudio/core/client/theme/DocTabLayoutPanel.java

```diff
@@ -237,7 +237,16 @@ public void moveTab(int index, int delta)
    @Override
    public void selectTab(int index)
    {
-      super.selectTab(index);
+      // select the tab, but don't fire events (we need to fire our own)
+      super.selectTab(index, false);
+      
+      // determine whether focus is currently inside the container
+      boolean focused = DomUtils.contains(getElement(), DomUtils.getActiveElement());
```

Comment:
This would be nice to have as a `DomUtils.containsFocus()` method, will be needing that for some related work I'll be doing.